oteljava/trace/src/main/scala/org/typelevel/otel4s/oteljava/trace/SpanBuilderImpl.scala

```diff
@@ -42,13 +42,13 @@ private[oteljava] final case class SpanBuilderImpl[F[_]: Sync] private (
     meta: InstrumentMeta[F],
     runner: SpanRunner[F],
     scope: TraceScope[F, Context],
-    stateModifier: SpanBuilder.State => SpanBuilder.State
+    stateModifiers: Vector[SpanBuilder.State => SpanBuilder.State]
```

Comment:
this gets into optimisation at this point, but I genuinely think `scala.collection.immutable.Queue` is better for this. quite a bit less copying (especially for smaller numbers of modifiers), with a single `List` reversal upon building the state